quantization/optimal_quantization.py

```diff
@@ -0,0 +1,171 @@
+"""
+Optimal quantization of the empirical distribution of a dataset -
+
+The algorithm finds an optimal aproximation of the dataset by a
+smaller number n of points, and yields a K-means type clustering.
+The n points of the approximation, weighted by the masses of their
+Voronoi cells, define a discrete distribution that best approximates,
+in the sense of the Wasserstein distance and at fixed support size n,
+the empirical distribution of the dataset. The Voronoi diagram associated
+to these points yields a K-means type clustering of the dataset.
+"""
+
+import numpy as np
+from geomstats.hypersphere import HypersphereMetric
+
+
+TOLERANCE = 1e-5
+IMPLEMENTED = ('S1', 'S2')
+
+
+def sample_from(points, size=1):
```

Comment:
There might be a function doing this already, something like: https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.choice.html
?

Reply:
It seems like `numpy.random.choice` takes only 1D arrays as entries, and so it cannot be directly used to sample from a set of points in 2 or more dimensions. I can use it to choose an index, but I don't think it would be very different from using `numpy.random.randint` ?